src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -599,7 +622,12 @@ namespace xclcpuemhal2 {
         delete []memTopology;
         memTopology = NULL;
       }
-
+      if (mIsKdsSwEmu)
+      {
+        mCore = new exec_core;
+        mSWSch = new SWScheduler(this);
+        mSWSch->init_scheduler_thread();
+      }
```

Comment:
This comment is not related to this change, but this function xclLoadXclBin is much too long.  It uses raw new (f.x memToplogy) and deletes.   It is not exception safe.

Reply:
sure. noted it.